File: packages/astro-polyglot/tests/fixtures/python/typed_module.py
"""A minimal typed Python module used as a conformance fixture."""

def greet(name: str) -> str:
    """Return a friendly greeting."""
    return f"Hello, {name}!"


class Calculator:
    """A simple calculator with typed methods."""

    def add(self, a: int, b: int) -> int:
        """Add two integers."""
        return a + b

    def divide(self, a: float, b: float) -> float:
        """Divide two floats."""
        if b == 0.0:
            raise ValueError("Cannot divide by zero")
        return a / b

File: packages/astro-polyglot/tests/fixtures/typescript/typed_module.ts
/** A minimal typed TypeScript module used as a conformance fixture. */

/** Return a friendly greeting. */
export function greet(name: string): string {
  return `Hello, ${name}!`;
}

/** A simple calculator with typed methods. */
export class Calculator {
  /** Add two integers. */
  add(a: number, b: number): number {
    return a + b;
  }

  /** Divide two floats. */
  divide(a: number, b: number): number {
    if (b === 0) {
      throw new Error("Cannot divide by zero");
    }
    return a / b;
  }
}